### PR TITLE
Add opacity prop for Box

### PR DESCRIPTION
**Pull request by jennyscript**
## TODO

- [x] Documentation
- [x] Tests
- [ ] Experimental evidence (required for Masonry changes)
- [ ] Accessibility checkup

### CODE REVIEW

File: packages/gestalt/src/Box.js [hunk 1]
```diff
@@ -190,6 +190,8 @@ type PropType = {
   minHeight?: number | string,
   minWidth?: number | string,
 
+  opacity?: 0 | 0.1 | 0.2 | 0.3 | 0.4 | 0.5 | 0.6 | 0.7 | 0.8 | 0.9 | 1,
```
Comment (christianvuerings) [inline packages/gestalt/src/Box.js]: @jennyscript let's not forget about our propTypes

File: packages/gestalt/src/Box.js [hunk 2]
```diff
@@ -307,6 +309,14 @@ const lgPaddingX = bind(rangeWithoutZero('lgPaddingX'), whitespace);
 const lgPaddingY = bind(rangeWithoutZero('lgPaddingY'), whitespace);
 const lgPadding = union(lgPaddingX, lgPaddingY);
 
+const opacityMap = mapClassName(name => styles[name]);
+const opacity = val => {
+  if (val > 0 && val < 1) {
```
Comment (christianvuerings) [inline packages/gestalt/src/Box.js]: @jennyscript Can we avoid this fork?

`String(val).replace('.', '')`

```
[0, 0.3, 1].map(val => String(val).replace('.', ''))
// ["0", "03", "1"]
```

Comment (jennyscript) [reply packages/gestalt/src/Box.js]: Range only takes number arguments. I'll see if one of the string compatible ones could use that form

Comment (jennyscript) [reply packages/gestalt/src/Box.js]: I tried a couple hacky workarounds, but this is the most clear option so I'd prefer to keep it

**Pull request merged**